lib/Server.js

```diff
@@ -630,7 +635,21 @@ class Server {
       options.https.key = options.https.key || fakeCert;
       options.https.cert = options.https.cert || fakeCert;
 
-      if (!options.https.spdy) {
+      // options.https.spdy is here for slightly better backwards compatability,
+      // since a user will probably provide this deprecated option if they
+      // are expecting the spdy server to be used.
+      const isHttp2 = options.http2 || options.https.spdy;
+
+      // note that options.spdy never existed. The user was able
+      // to set options.https.spdy before, though it was not in the
+      // docs. Keep options.https.spdy if the user sets it for
+      // backwards compatability, but log a deprecation warning.
+      if (options.https.spdy) {
+        this.log.warn(
+          'Providing custom spdy server options is deprecated and will be removed in the next major version.'
+        );
+      } else {
```

Comment:
`spdy` should enable `h2` (don't do braking change, even when option doesn't exists in docs) 

Deprecation message is ok :+1:

Reply:
This is not a breaking change, if `options.https.spdy` is set by the user, it will not be changed at all currently: https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L650